src/fondant/explore.py

```diff
@@ -97,27 +76,106 @@ def run_explorer_app(  # type: ignore
         logging.info(f"Using remote base path: {base_path}")
 
         # Mount extra volumes to the container
-        if extra_volumes:
-            for volume in extra_volumes:
-                cmd.extend(
-                    ["-v", volume],
-                )
-
-        # add the image name
-        cmd.extend(
-            [
-                f"{shlex.quote(container)}:{shlex.quote(tag)}",
-            ],
-        )
+        volumes.extend(extra_volumes)
 
         # Add the remote base path as a container argument
-        cmd.extend(
-            ["--base_path", f"{base_path}"],
-        )
+        command = ["--base_path", base_path]
+
+    services = {
+        f"{_get_service_name(container)}": {
+            "command": command,
+            "volumes": volumes,
+            "ports": [f"{port}:8501"],
+            "image": f"{shlex.quote(container)}:{shlex.quote(tag)}",
+        },
+    }
+
+    return {
+        "name": "explorer_app",
+        "version": "3.8",
+        "services": services,
+    }
+
+
+# type: ignore
+def run_explorer_app(  # type: ignore  # noqa: PLR0913
+    base_path: str,
+    port: int = PORT,
+    container: str = CONTAINER,
+    output_path: str = OUTPUT_PATH,
+    tag: t.Optional[str] = None,
+    extra_volumes: t.Union[t.Optional[list], t.Optional[str]] = None,
+):  # type: ignore
+    """
+    Run an Explorer App in a Docker container.
+
+    Args:
+      base_path: the base path where the Explorer App will be mounted.
+      port: The port number to expose the Explorer App. Default is 8501.
+      container: The name of the Docker container. Default is "fndnt/data_explorer".
+      output_path: The path to the Docker Compose specification. Default is
+       ".fondant/explorer-compose.yaml".
+      tag: The tag/version of the Docker container. Default is "latest".
+      extra_volumes: Extra volumes to mount in containers. You can use the --extra-volumes flag
+      to specify extra volumes to mount in the containers this can be used:
+        - to mount data directories to be used by the pipeline (note that if your pipeline's
+            base_path is local it will already be mounted for you).
+        - to mount cloud credentials
+    """
+    os.makedirs(".fondant", exist_ok=True)
+
+    explorer_app_spec = _generate_explorer_spec(
+        base_path=base_path,
+        port=port,
+        container=container,
+        tag=tag,
+        extra_volumes=extra_volumes,
+    )
+
+    with open(output_path, "w") as outfile:
+        yaml.dump(explorer_app_spec, outfile, default_flow_style=False)
+
+    cmd = [
+        "docker",
+        "compose",
+        "-f",
+        output_path,
+        "up",
+        "--build",
+        "--pull",
+        "always",
+        "--remove-orphans",
+        "--detach",
+    ]
+
+    subprocess.call(cmd, stdout=subprocess.PIPE)  # nosec
```

Comment:
I think we should use [`check_call`](https://docs.python.org/3/library/subprocess.html#subprocess.check_call) here so we don't print the messages below if this fails (eg. when the docker daemon is not running).